File: 66.plus-one/solution.go
func plusOne(digits []int) []int {
    carry := 1
    for i := len(digits) - 1; i >= 0; i-- {
        digit := digits[i] + carry
        digits[i], carry = digit % 10, digit / 10
    }
    if carry > 0 {
        return append([]int{carry}, digits...)
    }
    return digits
}

File: 66.plus-one/solution.kt
class Solution {
    fun plusOne(digits: IntArray): IntArray {
        for (i in digits.lastIndex downTo 0) {
            if (digits[i] < 9) {
                digits[i]++
                return digits
            }
            digits[i] = 0
        }
        return IntArray(digits.size + 1).apply { this[0] = 1 }
    }
}